docs/cpp/inline-functions-cpp.md

```diff
@@ -202,62 +202,6 @@ Assuming coordinate manipulation is a relatively common operation in a client of
 
 - Return
 
-## Inline functions vs. macros
-
-Inline functions are similar to macros, because the function code is expanded at the point of the call at compile time. However, inline functions are parsed by the compiler, and macros are expanded by the preprocessor. As a result, there are several important differences:
```

Comment:
I think we should keep this paragraph.

It seems to me that only "Inline functions are similar to macros" needs to be improved. They are very different to me.

Reply:
I think we should also include the fact that inline substitution is not mandatory, it occurs at the compiler's discretion, in the list. Currently the wording is confusing. `Inline functions are similar to macros, because the function code is expanded at the point of the call at compile time.` sounds like the substitution always happens.

Reply:
@codeworm96 Your idea is also my idea.

Reply:
> It seems to me that only "Inline functions are similar to macros" needs to be improved. 

I agree. The statement is not that rigorous, and this causal relationship is also kind of far-fetched.

Reply:
@frederick-vs-ja 
>Inline functions are similar to macros, because the function code is expanded at the point of the call at compile time.

I think we can just delete this paragraph and keep the last one:

> inline functions are parsed by the compiler, and macros are expanded by the preprocessor

Reply:
Replace this paragraph with:

## Inline functions vs. macros

inline functions are parsed by the compiler, and macros are expanded by the preprocessor.

Reply:
The original wasn't good, and I'm not sure this is an improvement. 'Parsed by the compiler', although technically true, only indirectly conveys useful information. The crux of the matter that we are trying to get at is that the compiler, at its discretion, actually compiles the function and drops the code of the body into your code without the overhead of a function call, and with the benefit of actually going through the compiler so that language syntax and semantics are properly observed. 
I see what the original author was trying to do here, but I think orienting the discussion about what an inline function is by comparing them with macros hurts more than it helps. I'll take a whack at redoing this section and we can see if it's an improvement.